src/fsearch_window_actions.c

```diff
@@ -584,6 +586,30 @@ fsearch_window_action_open_folder(GSimpleAction *action, GVariant *variant, gpoi
     fsearch_window_action_open_generic(self, true, false);
 }
 
+static void
+fsearch_window_action_preview(GSimpleAction *action, GVariant *variant, gpointer user_data) {
+    FsearchApplicationWindow *self = user_data;
+    guint xid = 0;
+    GList *file_list = NULL;
+
+    const guint selected_rows = fsearch_application_window_get_num_selected(self);
+    if (selected_rows == 0) {
+        return;
+    }
+
+    fsearch_application_window_selection_for_each(self, prepend_full_path_to_list, &file_list);
+    file_list = g_list_reverse(file_list);
+
+    GtkWidget *toplevel = gtk_widget_get_toplevel(GTK_WIDGET(self));
+#ifdef GDK_WINDOWING_X11
+    GdkWindow *window = gtk_widget_get_window (toplevel);
+    if (GDK_IS_X11_WINDOW(window)) {
+        xid = gdk_x11_window_get_xid(gtk_widget_get_window(toplevel));
+    }
+#endif
+    fsearch_preview_call_show_file(file_list, xid, TRUE);
```

Comment:
It might be a better idea to pass the `path` of the first selected file instead of `file_list`, since `fsearch_preview_call_show_file()` doesn't need the full list anyway.